tests/utils/fixtures/fixtures.py

```diff
@@ -155,14 +157,18 @@ def qemu_finalizer_impl(conn):
                 conn.run("poweroff")
                 halt_time = time.time()
                 # Wait up to 30 seconds for shutdown.
-                while halt_time + 30 > time.time() and qemu.poll() is None:
+                while (
+                    halt_time + shutdown_timeout_s > time.time() and qemu.poll() is None
+                ):
                     time.sleep(1)
             except:
                 pass
 
             # Terminate qemu
             try:
-                qemu.terminate()
+                os.killpg(os.getpgid(qemu.pid), signal.SIGTERM)
+                time.sleep(shutdown_timeout_s)
+                os.killpg(os.getpgid(qemu.pid), signal.SIGKILL)
```

Comment:
Don't use unconditional sleeps. Every such sleep adds up time in the tests, and it's very annoying when running locally. In fact I would just remove `SIGKILL` altogether. When qemu is hanging, it's extremely likely that it's the program inside qemu which is hanging, not qemu itself. So it should always respect `SIGTERM`.

Reply:
how about now? we sleep max 1s if the process exited, and if not, you should get annoyed, because even -9 did not help. opinions?

Reply:
Yep, that's better, now that it's conditional.